spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java

```diff
@@ -203,17 +217,66 @@ public static class Consumer {
 		 */
 		private RegexSubscriptionMode regexSubscriptionMode = RegexSubscriptionMode.PersistentOnly;
 
+		/**
+		 * Dead letter policy to use.
+		 */
+		@Nullable
+		private DeadLetterPolicy deadLetterPolicy;
+
+		/**
+		 * Whether to auto retry messages.
+		 */
+		private Boolean retryEnable = false;
+
 		/**
 		 * Whether the consumer auto-subscribes for partition increase. This is only for
 		 * partitioned consumers.
 		 */
 		private Boolean autoUpdatePartitions = true;
 
+		/**
+		 * Interval of partitions discovery updates.
+		 */
+		private Duration autoUpdatePartitionsInterval = Duration.ofMinutes(1);
+
 		/**
 		 * Whether to replicate subscription state.
 		 */
 		private Boolean replicateSubscriptionState = false;
 
+		/**
+		 * Whether to include the given position of any reset operation like
+		 * {@link org.apache.pulsar.client.api.Consumer#seek(long) or
+		 * {@link Consumer#seek(MessageId)}}.
+		 */
+		private Boolean resetIncludeHead = false;
+
+		/**
+		 * Key-shared mode.
+		 */
+		// @Nullable
+		// private KeySharedMode keySharedMode;
```

Comment:
Had to remove KeySharedMode because of https://github.com/apache/pulsar/pull/18345